playground/frontend/lib/components/playground_run_or_cancel_button.dart

```diff
@@ -20,38 +20,38 @@ import 'package:flutter/widgets.dart';
 import 'package:playground_components/playground_components.dart';
 import 'package:provider/provider.dart';
 
-import '../modules/analytics/analytics_service.dart';
-import '../utils/analytics_utils.dart';
-
 class PlaygroundRunOrCancelButton extends StatelessWidget {
   const PlaygroundRunOrCancelButton();
 
   @override
   Widget build(BuildContext context) {
     return Consumer<PlaygroundController>(
-      builder: (context, playgroundController, child) {
-        final analyticsService = AnalyticsService.get(context);
-        final stopwatch = Stopwatch();
-        final exampleName = getAnalyticsExampleName(playgroundController);
-
-        return RunOrCancelButton(
-          playgroundController: playgroundController,
-          beforeCancel: () {
-            final exampleName = getAnalyticsExampleName(playgroundController);
-            analyticsService.trackClickCancelRunEvent(exampleName);
-          },
-          beforeRun: () {
-            stopwatch.start();
-            analyticsService.trackClickRunEvent(exampleName);
-          },
-          onComplete: () {
-            analyticsService.trackRunTimeEvent(
-              exampleName,
-              stopwatch.elapsedMilliseconds,
-            );
-          },
-        );
-      }
+      builder: (context, playgroundController, child) => RunOrCancelButton(
+        playgroundController: playgroundController,
+        beforeCancel: (runner) {
+          PlaygroundComponents.analyticsService.sendUnawaited(
```

Comment:
ToB and Playground have different events. We will have to implement bridge with ToB and Playground interfaces for abstractions and abstract + Google for implementations. Then we gain methods with named parameters which is a bit simpler interface than constructing events, except that we anyway have to construct events to store them and to manipulate their properties as a map.

So I prefer constructing event objects here.

Reply:
I agree with @Malarg that sendCancelRun makes a little more sense than sendUnawaited but is still unclear to me.  Should it be called sendRunOrCancel to be consistent with the RunOrCancelButton?

Reply:
`sendUnawaited` is to wrap the sending function with an `unawaited` call so we do not get the lint of a lost Future (just added to the doc comment). This is better than wrapping all analytics calls with explicit `unawaited`. So the first thing to get it is to think of it as just `send` which is simpler to understand.

`AnalyticsService.sendRunOrCancel` is not an option because the service must then be aware of the running state, but we aim to keep it thin, thus the separate methods.

I do not see a problem here because `RunOrCancelButton` has the separate callbacks so the three specific events can be fired there.